### ISSUE TITLE
Port to h3 built-in utils

### ISSUE THREAD

**Issue by pi0** [NozomuIkuta/h3-cors#10]
Hi dear @NozomuIkuta. I just had a chance to check your awesome work here!

I think, if you are fine, we can port major part of this utility, to h3 core. (open issue https://github.com/unjs/h3/issues/82)

Other than other utilities, regarding `defineCorsEventHandler` since h3 avoids making _wrapper_ as much as possible, perhaps we can make usage `handleCors(event)`. (similar to https://github.com/unjs/h3/blob/main/src/utils/cache.ts#L15). (1)

**Comment by nozomuikuta** [comment]
@pi0 

Hi, thank you for taking time to check my package!

Yes, providing CORS middleware from h3 package itself will definitely bring much better developer experience 😄

Is there anything that I can help for code migration?

**Comment by pi0** [comment]
If you can put a PR in, we could initiate migration.

Since your package has users, we might upgrade h3 peer dependency here and reexport utils and defineCorsEventHandler from core then.

**Comment by Baroshem** [comment]
Hey folks,

Adding myself here to the loop. Once this will me done/merged, please let me know as well. I will then update the security module accordingly so that the current nuxt users will remain their functionality :)

**Comment by nozomuikuta** [comment]
@pi0 @Baroshem 

Confirmed that CORS support had been released as [h3@1.5.0](https://github.com/unjs/h3/blob/main/CHANGELOG.md#v150), I've just deprecated this package on NPM registry with a message that leads users to h3 upgrade. 👍 

@Baroshem 

Should I move #9 to h3 repository, or can I close it?

---

P.S.

`h3-cors` was my very first OSS project and I never expected it to be used by such a lot of people.
It was definitely h3's and Nuxt Security module's popularity that made it possible.

Thank you so much!!

I will keep contribution to h3 core as a member of unjs organization. 🏃
